File: bundles/gitlab-eclipse-plugin/src/com/gitlab/eclipse/lsp/WebviewInfo.java
package com.gitlab.eclipse.lsp;

import java.util.List;

public class WebviewInfo {
	@Override
	public String toString() {
		return "WebviewInfo [id=" + id + ", title=" + title + ", uris=" + uris + "]";
	}

	private String id;
	private String title;
	private List<String> uris;

	public WebviewInfo(String id, String title, List<String> uris) {
		this.id = id;
		this.title = title;
		this.uris = uris;
	}

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public String getTitle() {
		return title;
	}

	public void setTitle(String title) {
		this.title = title;
	}

	public List<String> getUris() {
		return uris;
	}

	public void setUris(List<String> uris) {
		this.uris = uris;
	}
}

File: src/main/kotlin/com/gitlab/eclipse/lsp/WebviewInfo.kt
package com.gitlab.eclipse.lsp

/**
 * One entry of the language server's `webviewMetadata` response.
 *
 * Deserialized by Gson, which bypasses the constructor: [title] and [uris] can be `null` at runtime
 * despite their declared types, and every reader here has to allow for that.
 */
data class WebviewInfo(var id: String, var title: String, var uris: List<String>) {
  /**
   * Design §17, for the same reason as `WebviewEditorKey.toString`: a data class prints every
   * component, and [uris] holds the advertised webview URIs — the same value
   * `WebviewResolution.Resolved.uri` is taken from, one step upstream, and the one §17 keeps out of
   * the log because nothing guarantees it carries no authentication material. Nothing stringifies a
   * `WebviewInfo` today; this replaces the generated form rather than resting on that, because one
   * interpolation anywhere would be enough.
   *
   * Only `toString` changes. Gson reads and writes the fields reflectively, so serialization,
   * `equals` and `hashCode` are untouched.
   */
  override fun toString(): String = "WebviewInfo($id)"
}
